src/win/fs.c

```diff
@@ -1268,13 +1268,14 @@ void fs__mktemp(uv_fs_t* req, uv__fs_mktemp_func func) {
         len = strlen(req->path);
         wcstombs((char*) req->path + len - num_x, ep - num_x, num_x);
       }
-      break;
+      return;
     }
   } while (--tries);
 
-  if (tries == 0) {
-    SET_REQ_RESULT(req, -1);
-  }
+  SET_REQ_RESULT(req, -1);
+
+clobber:
+  (char) *(req->path) = '\0';
```

Comment:
This cast seems weird
```suggestion
  req->path[0] = '\0';
```